### PR TITLE
Unify labels in serverless components

### PR DESCRIPTION
**Pull request by moelsayed**
**Description**

Changes proposed in this pull request:
This PR slightly changes the labels on the serverless components to align more with k8s best practices. However, it doesn't make any changes to the selector labels. The selector labels are immutable, and that would break upgrades.

- Align labels for server components with https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/.
- ...
- ...

**Related issue(s)**

[#15099](https://github.com/kyma-project/kyma/issues/15099)

### CODE REVIEW

File: resources/serverless/values.yaml [hunk 1]
```diff
@@ -58,12 +58,11 @@ tests:
 global:
   domainName: "kyma.example.com"
   commonLabels:
-    app: '{{ template "name" . }}'
-    version: "{{ .Values.global.images.function_controller.version }}"
-    app.kubernetes.io/name: '{{ template "name" . }}'
+    app: "{{ .Release.Name }}"
+    app.kubernetes.io/name: "{{ .Release.Name }}"
     app.kubernetes.io/instance: "{{ .Release.Name }}"
+    app.kubernetes.io/component: '{{ template "fullname" . }}-controller'
```
Comment (dbadura) [inline resources/serverless/values.yaml:64]: What about just `controller`?

File: resources/serverless/charts/webhook/values.yaml [hunk 2]
```diff
@@ -8,10 +8,11 @@ image:
   pullPolicy: IfNotPresent
 
 commonLabels:
-  app: '{{ template "webhook.fullname" . }}'
-  app.kubernetes.io/name: '{{ template "webhook.fullname" . }}'
+  app: "{{ .Release.Name }}"
+  app.kubernetes.io/name: "{{ .Release.Name }}"
   app.kubernetes.io/instance: "{{ .Release.Name }}"
-  role: webhook
+  app.kubernetes.io/component: '{{ template "webhook.fullname" . }}'
```
Comment (dbadura) [inline resources/serverless/charts/webhook/values.yaml:14]: Maybe `webhook` is enough?

**Review by dbadura — COMMENTED** [review]
Do we need to have` instance` label` as we have one?

File: resources/serverless/charts/docker-registry/templates/deployment.yaml [hunk 3]
```diff
@@ -5,10 +5,12 @@ kind: Deployment
...
   name: {{ template "docker-registry.fullname" . }}
   labels:
-    app: {{ template "docker-registry.name" . }}
-    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
-    release: {{ .Release.Name }}
-    heritage: {{ .Release.Service }}
+    app: "{{ .Release.Name }}"
+    app.kubernetes.io/name: "{{ .Release.Name }}"
+    app.kubernetes.io/instance: "{{ .Release.Name }}"
+    app.kubernetes.io/component: '{{ template "docker-registry.name" . }}'
```
Comment (dbadura) [inline resources/serverless/charts/docker-registry/templates/deployment.yaml:11]: `docker-registry`

**Comment by dbadura** [comment]
We can change the pod selector labels if we change the name of deployment to `serverless-webhook`.
The reconciler will delete the old deployment and apply new one.

File: resources/serverless/values.yaml (hunk 1 above)
Comment (moelsayed) [reply resources/serverless/values.yaml:64]: I think that might collide with other components that could be using the same pattern. This way it's more specific.

File: resources/serverless/charts/webhook/values.yaml (hunk 2 above)
Comment (moelsayed) [reply resources/serverless/charts/webhook/values.yaml:14]: Same reasoning as above.

File: resources/serverless/charts/docker-registry/templates/deployment.yaml (hunk 3 above)
Comment (moelsayed) [reply resources/serverless/charts/docker-registry/templates/deployment.yaml:11]: also as above, but this one is mostly to stick to the pattern.

**Comment by moelsayed** [comment]
@dbadura did you confirm this behavior ?

**Comment by dbadura** [comment]
I tested it by `kyma` cli which uses reconciler and it worked.

File: resources/serverless/values.yaml (hunk 1 above)
Comment (dbadura) [reply resources/serverless/values.yaml:64]: I wouldn't say it collide.
If I want to get all k8s controllers, I am able to do that with that label.

Comment (dbadura) [reply resources/serverless/values.yaml:64]: Look at the docs: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels
The component is named just: `database`

Comment (moelsayed) [reply resources/serverless/values.yaml:64]: Controller deployment template has the `kyma-project.io/component: controller` which would suitable for matching controllers. Using the `app.kubernetes.io/component` label this way is more aligned with the kubernetes docs recommendation.

**Comment by moelsayed** [comment]
@dbadura I am unable to reproduce this. If I changed the webhook deployment name, the old deployment is not deleted when I upgrade to the new version

**Comment by moelsayed** [comment]
I am closing this PR for now, as this solution is not sufficient to achieve the acceptance criteria. We need to tackle this from the reconciler side.

**Pull request closed without merging**